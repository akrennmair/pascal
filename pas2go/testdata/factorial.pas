{$mode ISO}
program test;

function factorial(i : integer) : integer;
begin
    if i = 1 then
        factorial := 1
    else
        factorial := i * factorial(i - 1)
end;

begin
    writeln('10! = ', factorial(10))
end.
