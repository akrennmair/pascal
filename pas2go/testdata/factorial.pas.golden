package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var factorial func(i int) int
	factorial = func(i int) (factorial_ int) {
		if i == 1 {
			factorial_ = 1
		} else {
			factorial_ = i * factorial(i-1)
		}
		return
	}

	system.Writeln("10! = ", factorial(10))
}
